proxy.go

```diff
@@ -333,6 +333,11 @@ func (p *Proxy) handle(ctx *Context, conn net.Conn, brw *bufio.ReadWriter) error
 				// http.ReadRequest.
 				tlsconn := tls.Server(&peekedConn{conn, io.MultiReader(bytes.NewReader(b), bytes.NewReader(buf), conn)}, p.mitm.TLSForHost(req.Host))
 
+				if err := tlsconn.Handshake(); err != nil && p.mitm.TLSHandshakeErrorHandler != nil {
+					p.mitm.TLSHandshakeErrorHandler(req, err)
```

Comment:
Is "req" here the CONNECT request? Maybe document that this is the case in the public API.